test/integration/jobset_controller_test.go

```diff
@@ -426,3 +247,163 @@ func checkJobSetStatus(js *jobset.JobSet, condition jobset.JobSetConditionType)
 	}
 	return false, nil
 }
+
+func numExpectedJobs(js *jobset.JobSet) int {
+	expectedJobs := 0
+	for _, rjob := range js.Spec.Jobs {
+		expectedJobs += rjob.Replicas
+	}
+	return expectedJobs
+}
+
+func completeAllJobs(jobList *batchv1.JobList) {
+	ginkgo.By("completing all jobs")
+	for _, job := range jobList.Items {
+		job.Status.Conditions = append(job.Status.Conditions, batchv1.JobCondition{
+			Type:   batchv1.JobComplete,
+			Status: corev1.ConditionTrue,
+		})
+		gomega.Expect(k8sClient.Status().Update(ctx, &job)).Should(gomega.Succeed())
+	}
+}
+
+func completeJob(job *batchv1.Job) {
+	ginkgo.By(fmt.Sprintf("completing job: %s", job.Name))
+	job.Status.Conditions = append(job.Status.Conditions, batchv1.JobCondition{
+		Type:   batchv1.JobComplete,
+		Status: corev1.ConditionTrue,
+	})
+	gomega.Expect(k8sClient.Status().Update(ctx, job)).Should(gomega.Succeed())
+}
+
+func failJob(job *batchv1.Job) {
+	ginkgo.By(fmt.Sprintf("failing job: %s", job.Name))
+	job.Status.Conditions = append(job.Status.Conditions, batchv1.JobCondition{
+		Type:   batchv1.JobFailed,
+		Status: corev1.ConditionTrue,
+	})
+	gomega.Expect(k8sClient.Status().Update(ctx, job)).Should(gomega.Succeed())
+}
+
+func checkJobsRecreated(js *jobset.JobSet, expectedRestarts int) (bool, error) {
+	var jobList batchv1.JobList
+	if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
+		return false, err
+	}
+	// Check we have the right number of jobs.
+	if len(jobList.Items) != numExpectedJobs(js) {
+		return false, nil
+	}
+	// Check all the jobs restart counter has been incremented.
+	for _, job := range jobList.Items {
+		if job.Labels[jobset.RestartsLabel] != strconv.Itoa(expectedRestarts) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func checkNumJobs(ctx context.Context, js *jobset.JobSet) (int, error) {
+	var jobList batchv1.JobList
+	if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
+		return -1, err
+	}
+	return len(jobList.Items), nil
+}
+
+// Check one headless service per job was created successfully.
+func checkOneHeadlessSvcPerJob(js *jobset.JobSet) {
+	gomega.Eventually(func() (int, error) {
+		var svcList corev1.ServiceList
+		if err := k8sClient.List(ctx, &svcList, client.InNamespace(js.Namespace)); err != nil {
+			return -1, err
+		}
+		return len(svcList.Items), nil
+	}).Should(gomega.Equal(numExpectedJobs(js)))
+}
+
+// - 3 replicated jobs, 1 replica each
+func simpleJobSet() *jobset.JobSet {
+	return testing.MakeJobSet("js-succeed", ns.Name).
+		AddReplicatedJob(testing.MakeReplicatedJob("replicated-job-a").
```

Comment:
to simplify things and make sure we test cases where replicas > 1, lets have two replicatedJobs here, one with replicas = 1 and one with replicas > 1 and use that in all tests.

Reply:
Ok the params for this base test jobset are now:

- one replicated job with 1 replica
- one replicated job with 3 replicas and pod DNS hostnames enabled